org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/IFilterOutput.java

```diff
@@ -41,4 +43,6 @@
 	 */
 	void merge(AbstractInsnNode i1, AbstractInsnNode i2);
 
+	void replace(AbstractInsnNode i, List<AbstractInsnNode> branches);
```

Comment:
I wander what exactly this method does... can you please add some JavaDoc?

Reply:
@marchof I was about to ask your advice about how to document it based on what its implementation does :wink:

Something like

```
	/**
	 * Marks instruction that should be considered as having branches into the
	 * given instructions during computation of coverage.
	 * 
	 * @param i
	 *            instruction to be replaced
	 * @param branches
	 *            targets of branches
	 */
```

?

Reply:
Ok, so it does actually not replace the instruction but its branches. In this case I would rename the method and parameters to

    replaceBranches(origin, targets)

Also maybe a second sentence can clarify the behavior: "The original target branches of the `origin` instruction are ignored."

Reply:
As like before this change term "branch" is fragile - we consider that each instruction has at least one branch and will be counted as covered when at least one branch is covered:

https://github.com/jacoco/jacoco/blob/d0e461a331989925776199978d22d8db82ceba03/org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java#L381-L388

So in some sense this replaces instruction and not just its branches:

https://github.com/jacoco/jacoco/blob/cc432204049c41720c8d8b2b64610fccc43d0380/org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java#L389-L411

and while we don't have use-case and don't need this, even

```
output.replace(original, Collections.singletonList(replacement))
```

works.

Thus not sure about renaming of method.
Renaming of second parameter into `targets` sounds good.
And for first one feel that `original` sounds better than `origin`.

----

I like the idea to add second statement :thumbsup: with slight rewording

> The original target branches of the **original** instruction **will be** ignored.

Reply:
Well, at least in the Analyzer I used the term "branch" as a synonym for a edge in the control flow graph. But with the JavaDoc we can leave the method as is.

Reply:
@marchof I added
* Javadoc to `IFilterOutput`
* tests to `MethodAnalyzerTest`
* changelog entry